lte/cloud/go/services/subscriberdb/storage/per_sub_digest.go

```diff
@@ -0,0 +1,121 @@
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package storage
+
+import (
+	lte_protos "magma/lte/cloud/go/protos"
+	"magma/lte/cloud/go/services/subscriberdb/protos"
+	"magma/orc8r/cloud/go/blobstore"
+	"magma/orc8r/cloud/go/mproto"
+	"magma/orc8r/cloud/go/storage"
+	merrors "magma/orc8r/lib/go/errors"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/pkg/errors"
+)
+
+const (
+	PerSubDigestBlobstoreType = "per_sub_digest"
+	// NOTE: perSubDigestBlobstoreNetworkKey is the placeholder network for the
+	// per-sub digest blobstore, since the actual network of each blob of per-sub
+	// digests is used as the key of the blob.
+	perSubDigestBlobstoreNetworkKey = "per_sub_digest_network_internal"
+)
+
+type PerSubDigestStore struct {
+	fact blobstore.BlobStorageFactory
+}
+
+func NewPerSubDigestStore(fact blobstore.BlobStorageFactory) *PerSubDigestStore {
+	return &PerSubDigestStore{fact: fact}
+}
+
+// GetDigest returns a list of per-subscriber digests of a network, ordered by their subscriber ID.
+func (l *PerSubDigestStore) GetDigest(network string) ([]*lte_protos.SubscriberDigestWithID, error) {
+	store, err := l.fact.StartTransaction(&storage.TxOptions{ReadOnly: true})
+	if err != nil {
+		return nil, errors.Wrapf(err, "error starting transaction")
+	}
+	defer store.Rollback()
+
+	blob, err := store.Get(perSubDigestBlobstoreNetworkKey, storage.TypeAndKey{Type: PerSubDigestBlobstoreType, Key: network})
+	if err == merrors.ErrNotFound {
+		// If per-sub digests for this network is not set yet, return empty list
+		return []*lte_protos.SubscriberDigestWithID{}, nil
+	}
+	if err != nil {
+		return nil, errors.Wrapf(err, "get per-sub digests of network %+v from blobstore", network)
+	}
+
+	perSubDigests := &protos.SubscriberDigestWithIDs{}
+	err = mproto.Unmarshal(blob.Value, perSubDigests)
+	if err != nil {
+		return nil, errors.Wrapf(err, "deserialize per-sub digests")
+	}
+
+	return perSubDigests.Digests, store.Commit()
+}
+
+// SetDigest creates or updates the per-subscriber digests of a network.
+func (l *PerSubDigestStore) SetDigest(network string, digests []*lte_protos.SubscriberDigestWithID) error {
+	store, err := l.fact.StartTransaction(&storage.TxOptions{ReadOnly: true})
+	if err != nil {
+		return errors.Wrapf(err, "error starting transaction")
+	}
+	defer store.Rollback()
+
+	// The sorted list of per-sub digests are serialized to be stored as a single blob.
+	// This is to preserve the sorted order of the digests, and to decrease the frequency
+	// of writes to the store.
+	blobValue := &protos.SubscriberDigestWithIDs{Digests: digests}
+	blobValueSerialized, err := mproto.MarshalDeterministic(blobValue)
```

Comment:
No need to use deterministic serialization here. Instead, use https://github.com/magma/magma/blob/509e5b4bcdb68b0021044fb1dc8cf47554446b8f/orc8r/lib/go/protos/marshaler.go#L27

Then you can remove the mproto.Unmarshal fn you added, along with the accompanying tests